code/z_adventurezones/lavamoon.dm

```diff
@@ -853,19 +867,18 @@ var/global/iomoon_blowout_state = 0 //0: Hasn't occurred, 1: Moon is irradiated
 				//target_marker = image('icons/misc/worlds.dmi', "boss_marker")
 				//target_marker.layer = FLY_LAYER
 
-				rotors = new /obj/iomoon_boss/rotor (locate(src.x - 2, src.y - 2, src.z))
-				rotors.core = src
-
-				base = new /obj/iomoon_boss/base (rotors.loc)
+				base = new /obj/iomoon_boss/base (locate(src.x - 2, src.y - 2, src.z))
 				base.core = src
 
+				top = new /obj/iomoon_boss/top (base.loc)
+
 				zapMarker = new /obj/iomoon_boss/zap_marker (src)
 
 				for (var/obj/iomoon_boss/bot_spawner/spawner in range(src, 10))
 					spawners += spawner
 
 		attackby(obj/item/I, mob/user as mob)
-			if (!I.force || active != 1)
+			if (!I.force || !IOCORE_ONLINE)
```

Comment:
!IOCORE_ONLINE will still pass when IOCORE_ONLINE is set to -1, it really needs multiple defines as it has three states, off, active and dead